python-threatexchange/threatexchange/cli/main.py

```diff
@@ -325,6 +331,10 @@ def inner_main(
     settings, extensions = _get_settings(config, state_dir)
     ap = get_argparse(settings)
     namespace = ap.parse_args(args)
+    if namespace.factory_reset:
```

Comment:
Without a way to garbage collect, folks like me factory resetting 2-3 times per session will fill up my directory. Additionally, some collabs (with TMKs in them) are gigabytes large, and so could fill up over time.

A solution that might work is to move it the /tmp/ directory, but since there is sensitive data in the state, I'd want to properly chmod it for safety, which requires some research and to make sure there's not any specific attacks still possible.

Rather than do that, I'll just continue to delete for now - it's pretty hard to accidentally hit this command (it must be fully typed out, only from the root command)